Applications/Spire/Source/Ui/ComboBox.cpp

```diff
@@ -39,7 +51,7 @@ ComboBox::ComboBox(std::shared_ptr<QueryModel> query_model,
       m_has_autocomplete_selection(false),
       m_current_connection(m_current->connect_update_signal(
         std::bind_front(&ComboBox::on_current, this))) {
-  m_input_box = new TextBox();
+  m_input_box = input_box_builder();
```

Comment:
If you don't need a function then use a plain value. As discussed in our meeting, a ```function``` is used only when the return value depends on side-effects or some combination of arguments.